### PR TITLE
Stop promoting unreleased.rst.

### PR DESCRIPTION
**Pull request by Zimmi48**

### CODE REVIEW

File: doc/dune [hunk 1]
```diff
@@ -1,8 +1,5 @@
 (rule
  (targets unreleased.rst)
- ; We need to keep this as the sphinx build still requires this file
- ; in-tree, to be fixed hopefully soon
```
Comment (SkySkimmer) [inline doc/dune:4]: So what changed?

Comment (Zimmi48) [reply doc/dune:4]: I'm not sure, but this comment is quite old, so probably the build consolidation PR. I also do not know what it meant exactly (cc @ejgallego).

Comment (ejgallego) [reply doc/dune:4]: Likely this was indeed from a time we had options to run sphinx both from make and from dune, should not be needed now.

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
Refman build seems broken by an unrelated issue.

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**